### PR TITLE
fixed: we need a separate bool to track if partitioning is enabled

### PR DESCRIPTION
**Pull request by akva2**
empty myElms is not good enough - we might own 0 elements
on the patch

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/LR/ASMu3D.C [hunk 1]
```diff
@@ -2498,6 +2501,7 @@ void ASMu3D::generateThreadGroupsFromElms (const IntVec& elms)
   for (int elm : elms)
     if (this->getElmIndex(elm+1) > 0)
       myElms.push_back(this->getElmIndex(elm+1)-1);
+  hasElms = true;
```
Comment (kmokstad) [inline src/ASM/LR/ASMu3D.C:2504]: An alternative (more elegant?, at least with less footprint) solution could be to add, e.g. 0 or -1 as a value in myElms here, such that it is never empty if we do generateThreadGroupsFromElms and the test in ASMu3D::integrate will still work.

**Review by kmokstad — CHANGES_REQUESTED** [review]
This would apply for ASMu2D also I persume?
Consider my alternative solution(here I'm not 100% sure about the logic).
At least, clean up the confusing myElms member in both ASMbase and ASMu3D.

File: src/ASM/LR/ASMu3D.C (hunk 1 above)
Comment (kmokstad) [reply src/ASM/LR/ASMu3D.C:2504]: The you don't need the hasElms member

File: src/ASM/LR/ASMu3D.h [hunk 2]
```diff
@@ -632,6 +632,7 @@ class ASMu3D : public ASMLRSpline, public ASM3D
   ThreadGroups threadGroups; //!< Element groups for multi-threaded assembly
   ThreadGroups projThreadGroups; //!< Element groups for multi-threaded assembly - projection basis
   IntVec       myElms;       //!< Elements on patch - used with partitioning
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/ASM/LR/ASMu3D.h:634]: yeah. in my original code there was no myElms member in ASMbase, they were only stored down in the DomainDecomposition member. I added it after your review comment, and then forgot to remove it in ASMu3D.
I will remove this first, before I redo the rest of this pr, https://github.com/OPM/IFEM/pull/443

File: src/ASM/LR/ASMu3D.C (hunk 1 above)
Comment (akva2) [reply src/ASM/LR/ASMu3D.C]: Okay, did this instead.

**Comment by akva2** [comment]
This applies to all ASMs. Just did not include it initially since I wanted review comments first. It's applied everywhere now.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Review by kmokstad — APPROVED** [review]
:+1:

**Pull request merged**